### PR TITLE
Filter inactive penalties processing by slashed

### LINKED ISSUE #5897: Check if validator was slashed before apply inactivity penalty
# 🐞 Bug Report

### Description

Discovered in Schlesi testnet, to process inactivity penalty, the node needs to check if the validator was slashed, in which case it would also get a finality penalty. 
https://github.com/prysmaticlabs/prysm/blob/master/beacon-chain/core/epoch/precompute/reward_penalty.go#L119

Note that this is a consensus breaking change so the fix has to go into `v0.12` branch

### PR DESCRIPTION
**Pull request by terencechain**
**What type of PR is this?**
> Bug fix

**What does this PR do? Why is it needed?**
This fixes a penalty processing bug, it ensures inactive penalties is only applied to validators who don't attest for the previous epoch target or are slashed. 

**Which issues(s) does this PR fix?**

Fixes #5897

**Other notes for review**
Don't merge this into master

### CODE REVIEW

File: beacon-chain/core/epoch/precompute/reward_penalty.go [hunk 1]
```diff
@@ -116,7 +116,7 @@ func attestationDelta(state *stateTrie.BeaconState, pBal *Balance, v *Validator)
 	finalityDelay := e - finalizedEpoch
 	if finalityDelay > params.BeaconConfig().MinEpochsToInactivityPenalty {
 		p += params.BeaconConfig().BaseRewardsPerEpoch * br
-		if !v.IsPrevEpochTargetAttester {
+		if !v.IsPrevEpochTargetAttester && !v.IsSlashed {
```
Comment (0xKiwi) [inline beacon-chain/core/epoch/precompute/reward_penalty.go:119]: Shouldn't this slashing check be inversed? Why would we punish validators that aren't slashed?

Comment (terencechain) [reply beacon-chain/core/epoch/precompute/reward_penalty.go:119]: This is inactive penalty

File: beacon-chain/core/epoch/precompute/reward_penalty.go [hunk 2]
```diff
@@ -116,7 +116,7 @@ func attestationDelta(state *stateTrie.BeaconState, pBal *Balance, v *Validator)
 	finalityDelay := e - finalizedEpoch
 	if finalityDelay > params.BeaconConfig().MinEpochsToInactivityPenalty {
 		p += params.BeaconConfig().BaseRewardsPerEpoch * br
```
Comment (0xKiwi) [inline beacon-chain/core/epoch/precompute/reward_penalty.go:118]: Can you make `BaseRewardsPerEpoch` a local var? It's used like 3 or 4 times in this function.

File: beacon-chain/core/epoch/precompute/reward_penalty.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/core/epoch/precompute/reward_penalty.go:119]: You are right. It should be `|| v.IsSlashed` Thanks!

File: beacon-chain/core/epoch/precompute/reward_penalty.go [hunk 3]
```diff
@@ -115,8 +114,8 @@ func attestationDelta(state *stateTrie.BeaconState, pBal *Balance, v *Validator)
 	finalizedEpoch := state.FinalizedCheckpointEpoch()
 	finalityDelay := e - finalizedEpoch
 	if finalityDelay > params.BeaconConfig().MinEpochsToInactivityPenalty {
-		p += params.BeaconConfig().BaseRewardsPerEpoch * br
-		if !v.IsPrevEpochTargetAttester {
+		p += bestRewardsPerEpoch * br
+		if !v.IsPrevEpochTargetAttester || v.IsSlashed {
```
Comment (prestonvanloon) [inline beacon-chain/core/epoch/precompute/reward_penalty.go:118]: Can you add some comments please?

File: beacon-chain/core/epoch/precompute/reward_penalty_test.go [hunk 4]
```diff
@@ -117,20 +125,21 @@ func TestAttestationDeltaPrecompute(t *testing.T) {
...
 		// Base rewards for getting source right
 		wanted := attestedBalance*base/totalBalance +
-			bp.PrevEpochTargetAttesters*base/totalBalance +
-			bp.PrevEpochHeadAttesters*base/totalBalance
+			bp.PrevEpochTargetAttested*base/totalBalance +
+			bp.PrevEpochHeadAttested*base/totalBalance
 		// Base rewards for proposer and attesters working together getting attestation
 		// on chain in the fatest manner
 		proposerReward := base / params.BeaconConfig().ProposerRewardQuotient
-		wanted += (base - proposerReward) * params.BeaconConfig().MinAttestationInclusionDelay
```
Comment (0xKiwi) [inline beacon-chain/core/epoch/precompute/reward_penalty_test.go:133]: This params is being reused in a foor loop, can you move it out? Same for the proposer reward one.

**Pull request merged**